File: plugins/battery/battery.30s.js
#!/usr/bin/env node
// battery.30s.js
const { execSync } = require('child_process');

function getBattery() {
    try {
        const out = execSync('crossbar battery --json', { encoding: 'utf8', timeout: 5000 });
        return JSON.parse(out);
    } catch { return null; }
}

const data = getBattery();
if (!data || data.level === undefined || data.level === null) {
    console.log("🔋 --");
    console.log("---");
    console.log("No battery detected");
    process.exit(0);
}

const { level, charging } = data;
let icon = "🔋", color = "green";

if (charging) {
    icon = "⚡"; color = "blue";
} else if (level <= 20) {
    icon = "🪫"; color = "red";
} else if (level <= 50) {
    color = "yellow";
}

console.log(`${icon} ${level}% | color=${color}`);
console.log("---");
console.log(`Battery Level: ${level}%`);
console.log(`Status: ${charging ? 'Charging' : 'Discharging'}`);
console.log("---");
console.log("Refresh | refresh=true");

File: plugins/battery/battery.30s.lua
-- battery.30s.lua
-- Battery status monitor using embedded Lua interpreter
-- Uses Universal Crossbar API

local bat = crossbar.battery()

if bat and bat.available then
    local icon = "🔋"
    local color = "green"
    
    if bat.charging then 
        icon = "⚡"
        color = "blue"
    elseif bat.level <= 20 then
        icon = "🪫"
        color = "red"
    elseif bat.level <= 50 then
        color = "yellow"
    end
    
    print(icon .. " " .. bat.level .. "% | color=" .. color)
    print("---")
    print("Battery Level: " .. bat.level .. "%")
    print("Status: " .. (bat.charging and "Charging" or "Discharging"))
    print("---")
    print("Refresh | refresh=true")
else
    print("🔋 --")
    print("---")
    print("No battery detected")
end
